DEVC-579 Add FireAngel Thermoptek ZST-630 Smoke Alarm/Detector to generic smoke alarm DTH

MAblewicz opened the pull request:
Add FireAngel ZST-630 Smoke Detector/Alarm support to zwave-basic-smoke-alarm.groovy

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Add FireAngel ZST-630 Smoke Detector/Alarm support

Commit:
unnecessary whitespace removal

Commit:
DEVC-579
Device name set to FireAngel ZST-630 Thermoptek Smoke Alarm/Smoke Detector
according to Z-wave alliance

greens reviewed (changes requested):
This will break the First Alert functionality. Because you will now be parsing all Command Class 0x71 (alarm/notification) events as v3, the alarm v2 handler that the First Alert was using will never be called.

You'll need to handle all events as notification v3, but include some special handling to distinguish the First Alert's Alarm v2 events.

You're also missing anything that would clear the "tested" event. Presumably they have overloaded the 0x00 event to be an "all clear" event that clears both smoke and test.

Commit:
DEVC-580
Post Review fixes
Added alarm version support depending on device

MAblewicz commented:
Thanks for suggestion. I will add code selecting alarm version based on Manufacturer specific data.
As for clearing tested event it is the device sends 0x00 when i press test button to neutralize test alarm. Also according to  http://z-wave.sigmadesigns.com/wp-content/uploads/2016/08/SDS12652-13-Z-Wave-Command-Class-Specification-N-Z.pdf   page 84-85 there is no event  for test clear definied  for Notification Type 0x01.

mckeed commented on devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:
```diff
@@ -61,12 +68,31 @@ def updated() {
+			break
+		default: alarmVersion = 2
+	}
+	return alarmVersion
```
v3 is backwards compatible to v2. You shouldn't be checking the msr to determine the version, just integrate the parsing for First Alert into the v3 handler.

MAblewicz replied:
I will merge some code from AlarmReport  to NotificationReport  (of course including field name changes due Notification repacement of Alarm )

mckeed commented on devicetypes/smartthings/zwave-basic-smoke-alarm.src/README.md:
```diff
@@ -22,18 +23,62 @@ Works with:
+
+Tap My Home 
+Under Things, tap Add a Thing at the bottom of your Things list
+The app will say Looking for devices�
```
Not a big deal, but It's not great to be adding documentation for the mobile app to these comments. This UI regularly changes and we don't want a bunch of outdated comments all over the place. Just say "put the hub in Add Device mode" or something like that.

MAblewicz replied:
Ok i will shorten the steps

mckeed commented on devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:
```diff
@@ -20,6 +20,8 @@ metadata {
 		fingerprint deviceId: "0xA100", inClusters: "0x20,0x80,0x70,0x85,0x71,0x72,0x86"
 		fingerprint mfr:"0138", prod:"0001", model:"0001", deviceJoinName: "First Alert Smoke Detector"
+		//zw:S type:0701 mfr:026F prod:0001 model:0001 ver:1.07 zwv:4.24 lib:03 cc:5E,86,72,5A,73,80,71,85,59,84 role:06 ff:8C01 ui:8C01
+		fingerprint mfr: "026F ", prod: "0001", model: "0001", deviceJoinName: "FireAngel Thermoptek ZST-630 Smoke Alarm/Detector"
```
Just "FireAngel Thermoptek Smoke Alarm" would be better imo

MAblewicz replied:
Sure i will change name to it

mckeed commented on devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:
```diff
@@ -61,12 +68,31 @@ def updated() {
 	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
The one-hour checkin is specific to the First Alert models. You'll need to determine if this device has a regular check-in and if not, use the 8 hour checkInterval that is default for sleepy z-wave devices.

In that case, make 8 hours the default, and set to the shorter 2 hours if the device is the FirstAlert model (use zwaveInfo.mfr to determine manufacturer)

greens commented on devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:
```diff
@@ -195,3 +223,35 @@ def zwaveEvent(physicalgraph.zwave.Command cmd, results) {
+		switch (cmd.event) {
+			case 0x00:
+			case 0xFE:
+				createSmokeEvents("smokeClear", results)
```
Based on the notification command class spec, 0xFE means "unknown command". 0x00 means "previous events cleared". So this should be adjusted so that type 1 event 0 creates both a "smokeClear" and a "testedClear" event.

I don't think 0xFE should do anything at all.

greens replied:
I realize now that "smokeClear" and "testClear" are functionally equivalent. So ignore me.

Commit:
Devc 579 Add FireAngel Thermoptek ZST-630 Smoke Alarm/Detector to generic smoke alarm DTH
Post Review fixes
Add First Alert support for Alarm Command Class version 3
Adjust healt check intervall depering on device
FireAngel name change

Commit:
Devc 579 Add FireAngel Thermoptek ZST-630 Smoke Alarm/Detector to generic smoke alarm DTH
Correct Healt check info for FireAngel
